gantry/simulator/CMakeLists.txt

```diff
@@ -1,18 +1,67 @@
+# Load free for the posix port
+find_package(FreeRTOS)
+
 # Set up the include directories
 include_directories(
         .
+        ../firmware
         ${CMAKE_CURRENT_SOURCE_DIR}/../../include
+        ${FreeRTOS_SOURCE_DIR}/FreeRTOS/Source/include
+        ${FreeRTOS_SOURCE_DIR}/FreeRTOS/Source/portable/ThirdParty/GCC/Posix
+        ${FreeRTOS_SOURCE_DIR}/FreeRTOS/Source/portable/ThirdParty/GCC/Posix/utils
+
+)
 
+# Create the FreeRTOS source file list
+FILE(
+        GLOB FREERTOS_SOURCES
+        ${FreeRTOS_SOURCE_DIR}/FreeRTOS/Source/*.c
 )
 
+# Add the posix port files
+list(APPEND FREERTOS_SOURCES "${FreeRTOS_SOURCE_DIR}/FreeRTOS/Source/portable/MemMang/heap_3.c")
+list(APPEND FREERTOS_SOURCES "${FreeRTOS_SOURCE_DIR}/FreeRTOS/Source/portable/ThirdParty/GCC/Posix/utils/wait_for_event.c")
+list(APPEND FREERTOS_SOURCES "${FreeRTOS_SOURCE_DIR}/FreeRTOS/Source/portable/ThirdParty/GCC/Posix/port.c")
+
+
+# Create the freertos lib
+add_library(freertos-gantry STATIC ${FREERTOS_SOURCES})
+
+if("${CMAKE_HOST_SYSTEM_NAME}" STREQUAL "Linux")
```

Comment:
we probably should use [generator expressions](https://cmake.org/cmake/help/latest/manual/cmake-generator-expressions.7.html#conditional-expressions) for this